### PR TITLE
Add IGListDiffable remodel plugin source + documentation (a new guide)

### PR DESCRIPTION
**Pull request by calimarkus**
## Changes in this pull request

Adds the source for a remodel plugin to autogenerate models conforming to `IGListDIffable`.

### Checklist

- [x] All tests pass. Demo project builds and runs. (source untouched, purely additive)
- [x] I added remodel plugin tests + and a new guide explaining the installation + usage
- [ ] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by rnystrom** [comment]
So exciting! I'm really glad we're open sourcing this. 🎉

**Comment by jessesquires** [comment]
nice! 👍

File: Guides/Generating your models.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Generating your IGListDiffable models
+
+With the `IGListDiffable` plugin for [remodel by facebook](https://github.com/facebook/remodel), you can automatically generate models conforming to the `IGListDiffable`.
+
+This will automatically implement `hash`, `isEqual:` and `description`, as well as `diffIdentifier` and `isEqualToDiffableObject:` for you. Remodel is also capable to generate additional code, like conforming to `NSCoding` or additional Builder classes for your model object. It will make creating and updating models much easier, faster and safer.
+
+In `/remodel-plugin`, you can find the source files to build the `IGListDiffable` plugin locally.
+
+### 1. Remodel installation (if you don't have it yet)
```
Comment (jessesquires) [inline Guides/Generating your models.md]: nit: let's remove "(if you don't have it yet)"

File: Guides/Generating your models.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Please follow the installation instructions on https://github.com/facebook/remodel.
```
Comment (jessesquires) [inline Guides/Generating your models.md]: nit: how about the following:

> Please follow the [installation instructions](https://github.com/facebook/remodel).

File: Guides/Generating your models.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+tl;dr: Either clone the original repository, or use an npm installation. In the latter case you can run `which remodel-gen` to find out the path of your installation.
+
+### 2. Plugin installation (copy source files)
```
Comment (jessesquires) [inline Guides/Generating your models.md]: nit: let's remove "(copy source files)"

File: Guides/Generating your models.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  }
+  return result;
+}
+
+@end
+```
+
+### Documentation
+
+Please see the main remodel repository for additional documentation: https://github.com/facebook/remodel
```
Comment (jessesquires) [inline Guides/Generating your models.md]: nit: how about:

> Please see the main remodel repository for [additional documentation](https://github.com/facebook/remodel).

Comment (jessesquires) [reply Guides/Generating your models.md]: also, I think we should make this h2, `##`

File: Guides/Generating your models.md (hunk 2 above)
Comment (calimarkus) [reply Guides/Generating your models.md]: I wanted to be verbose about the other repo. But sure.

**Comment by rnystrom** [comment]
I'm gonna hold this PR for 3.3 so that I can wrap up and push 3.2 w/out too many last-minute changes that haven't baked publicly on `master`.

**Comment by calimarkus** [comment]
Okay! I figured I'll add an example too using the generated models, so we have a generated class to look at in the example project. I went for a list of top contributors + me :P

xcode

examples

example

**Comment by rnystrom** [comment]
I love this

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
@calimarkus now that 3.2 is out, lmk when this is ready to land. Also if you want to just rerun the example pod script, that should fix the conflicts.

**Comment by calimarkus** [comment]
okay this is good to go and doesn't touch the generated pod project. Why do we check that in at all?

**Comment by rnystrom** [comment]
@calimarkus The pod stuff? Our philosophy has always been that the repo should be buildable straight from download. Helps onboard new folks so they don't have to wrestle w/ build tools. Just download, build, & run.

**Comment by calimarkus** [comment]
I see - this doesn't affect the pods in any way, so this is still good to go :)

**Pull request closed without merging**